beacon-chain/node/node.go

```diff
@@ -609,13 +596,19 @@ func (b *BeaconNode) registerBlockchainService() error {
 		blockchain.WithSlashingPool(b.slashingsPool),
 		blockchain.WithP2PBroadcaster(b.fetchP2P()),
 		blockchain.WithStateNotifier(b),
-		blockchain.WithForkChoiceStore(b.forkChoiceStore),
 		blockchain.WithAttestationService(attService),
 		blockchain.WithStateGen(b.stateGen),
 		blockchain.WithSlasherAttestationsFeed(b.slasherAttestationsFeed),
 		blockchain.WithFinalizedStateAtStartUp(b.finalizedStateAtStartUp),
 		blockchain.WithProposerIdsCache(b.proposerIdsCache),
 	)
+
+	if !features.Get().DisableForkchoiceDoublyLinkedTree {
+		opts = append(opts, blockchain.WithForkChoiceStore(doublylinkedtree.New()))
```

Comment:
Perhaps avoid the double negation here?
```suggestion
	if features.Get().DisableForkchoiceDoublyLinkedTree {
		opts = append(opts, blockchain.WithForkChoiceStore(protoarray.New()))
```
But anyway I think it's cleaner to move the forkchoice structure away from `blockchain.cfg` and into the `blockchain.Service` structure. And initialize it from blockchain instead of here. I don't feel strongly about this though.

Reply:
I think this clearly shows this dependency, could we leave it in cfg for now? I can revisit it later too